src/merkle/merkle-objects.ts

```diff
@@ -157,7 +157,7 @@ export class Candidate implements CIDHolder {
       this._alreadyAnchored = true
     } else {
       this._cid = stream.tip
-      this._metadata = stream.metadata as StreamMetadata
+      this._metadata = stream.state.metadata as StreamMetadata
```

Comment:
```suggestion
      this._metadata = stream.state.next?.metadata ? stream.state.next.metadata : stream.state.metadata as StreamMetadata
```

Reply:
sigh.  Just checking the state.metadata would work for Models and MIDs, but for TileDocument we need to also check for state.next.metadata since TileDocument does actually allow changing controllers.